sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp

```diff
@@ -201,6 +201,40 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
         const SetDataLakeFileSystemAccessPolicyOptions& options
         = SetDataLakeFileSystemAccessPolicyOptions()) const;
 
+    /**
+     * @brief Renames a file. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param fileName The file that gets renamed.
+     * @param destinationFilePath The path of the file the source file is renaming to.
+     * @param options Optional parameters to rename a file
+     * @return Azure::Core::Response<Models::RenameDataLakeDataLakeFileResult> containing the
+     * information returned when renaming the file, including the DataLakeFileClient targeting the
+     * renamed file.
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::RenameDataLakeFileResult> RenameFile(
+        const std::string& fileName,
+        const std::string& destinationFilePath,
+        const RenameDataLakeFileOptions& options = RenameDataLakeFileOptions()) const;
+
+    /**
+     * @brief Renames a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param directoryName The directory that gets renamed.
+     * @param destinationDirectoryPath The destinationPath the source directory is renaming to.
+     * @param options Optional parameters to rename a directory.
+     * @return Azure::Core::Response<Models::RenameDataLakeDirectorySinglePageResult> containing the
+     * information returned when renaming the directory, including the DataLakeDirectoryClient
+     * targeting the renamed directory.
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::RenameDataLakeDirectorySinglePageResult>
```

Comment:
I prefer it returns a `DataLakeClient`

Reply:
How would you also return the `ContinuationToken` then?